gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/ExportPanel.java

```diff
@@ -401,16 +416,31 @@ private File buildFile(final boolean isCustomExport) {
 		return tempFile;
 	}
 
-	private String buildFileName() {
-		final String prefix = "gradebook_export";
+
+	private String buildFileName(final boolean customDownload) {
+		final String prefix = getString("importExport.download.filenameprefix");
 		final String extension = this.exportFormat.toString().toLowerCase();
-		String gradebookName = this.businessService.getGradebook().getName();
+		final String gradebookName = this.businessService.getGradebook().getName();
 
-		if (StringUtils.trimToNull(gradebookName) == null) {
-			return String.format("%s.%s", gradebookName, extension);
-		} else {
-			gradebookName = gradebookName.replaceAll("\\s", "_");
-			return String.format("%s-%s.%s", prefix, gradebookName, extension);
+		// File name contains the prefix
+		final List<String> fileNameComponents = new ArrayList<>();
+		fileNameComponents.add(prefix);
+
+		// Add gradebook name/site id to filename
+		if (StringUtils.trimToNull(gradebookName) != null) {
+			fileNameComponents.add(gradebookName.replaceAll("\\s", "_"));
 		}
+
+		// If custom download for all sections, append 'ALL' to filename
+		if (customDownload && (this.group == null || this.group.getId() == null)) {
+			fileNameComponents.add(getString("importExport.download.filenameallsuffix"));
+
+		// If group/section filter is selected, add group title to filename
+		} else if (this.group != null && this.group.getId() != null && StringUtils.isNotBlank(this.group.getTitle())) {
+			final String sanitizedGroupName = this.group.getTitle().replaceAll("[\\s\\/\\\\\\:\\\"\\\']", "_");
```

Comment:
@jonespm do we have any easy sanitizer in Validator?

Reply:
I don't see any common sanitizeFilename or cleanFilename method? I do see a few open issues around filenames with accents or UTF-8 characters though that are issues. We probably should have a cleaner method in kernel utils though. Maybe something like https://stackoverflow.com/a/26420820/3708872  This line seems like it might miss some characters?

Reply:
FilenameUtils.normalize?  

Or just replace any non a-zA-Z0-9 with nothing?

Reply:
Thanks @ottenhoff, @jonespm.  Have changed it to `[^A-Za-z0-9]` which is far cleaner.  Happy to tweak again if there's something better!

Reply:
It doesn't look like normalize does anything with those special characters.
"Normalizes a path, removing double and single dot path steps."

My only question about this regex is it seems like it might be an issue for i18n characters, most commonly accented characters which should be acceptable and also UTF-8 like Asian characters. (Which for sure will have these names)

"This works on a file name that uses only English letters. If the file is made up of only Chinese characters then you will strip everything out of it."

Regular expression cookbook suggests 

[\\/:"*?<>|]+

which is closer to your original regex but a little simpler. This is probably why it would be *nice* to have a util method.

https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch08s25.html

Reply:
Ah good points @jonespm.   I've just pushed a patch that introduces `Validator.cleanFilename` with the exclusions defined above (it follows the pattern in which those helpers are currently implemented). Thanks!